x/auth/types/codec.go

```diff
@@ -2,32 +2,31 @@ package types
 
 import (
 	"github.com/cosmos/cosmos-sdk/codec"
-	"github.com/cosmos/cosmos-sdk/x/auth/exported"
+	"github.com/cosmos/cosmos-sdk/codec/types"
 )
 
-// Codec defines the interface needed to serialize x/auth state. It must be
-// aware of all concrete account types.
-type Codec interface {
-	codec.Marshaler
-
-	MarshalAccount(acc exported.Account) ([]byte, error)
-	UnmarshalAccount(bz []byte) (exported.Account, error)
-
-	MarshalAccountJSON(acc exported.Account) ([]byte, error)
-	UnmarshalAccountJSON(bz []byte) (exported.Account, error)
-}
-
 // RegisterCodec registers the account interfaces and concrete types on the
 // provided Amino codec.
 func RegisterCodec(cdc *codec.Codec) {
-	cdc.RegisterInterface((*exported.ModuleAccountI)(nil), nil)
-	cdc.RegisterInterface((*exported.GenesisAccount)(nil), nil)
-	cdc.RegisterInterface((*exported.Account)(nil), nil)
-	cdc.RegisterConcrete(&BaseAccount{}, "cosmos-sdk/Account", nil)
+	cdc.RegisterInterface((*ModuleAccountI)(nil), nil)
+	cdc.RegisterInterface((*GenesisAccount)(nil), nil)
+	cdc.RegisterInterface((*AccountI)(nil), nil)
+	cdc.RegisterConcrete(&BaseAccount{}, "cosmos-sdk/BaseAccount", nil)
 	cdc.RegisterConcrete(&ModuleAccount{}, "cosmos-sdk/ModuleAccount", nil)
 	cdc.RegisterConcrete(StdTx{}, "cosmos-sdk/StdTx", nil)
 }
 
+// RegisterInterface associates protoName with AccountI interface
+// and creates a registry of it's concrete implementations
+func RegisterInterfaces(registry types.InterfaceRegistry) {
+	registry.RegisterInterface(
+		"cosmos_sdk.auth.v1.AccountI",
+		(*AccountI)(nil),
+		&BaseAccount{},
+		&ModuleAccount{},
+	)
```

Comment:
q: if I want to add another account type? do I register is as follows? @aaronc :

```go
	registry.RegisterInterface(
		"cosmos_sdk.auth.v1.AccountI",
		(*AccountI)(nil),
		&EthermintAccount{},
	)
```

Reply:
asking because the interface will already registered if the auth is called first

Reply:
you can just use `RegisterImplementations`. the name in `RegisterInterface` is just informational. maybe we should get rid of the ability to add implementations with that method so that it's less confusing

Reply:
yeah, it's definitely confusing